robotiq_s_model_articulated_gazebo_plugins/src/RobotiqHandPlugin.cpp

```diff
@@ -0,0 +1,888 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+/*
+    This file has been modified from the original, by Devon Ash
+*/ 
+
+#include <robotiq_s_model_articulated_msgs/SModelRobotInput.h>
+#include <robotiq_s_model_articulated_msgs/SModelRobotOutput.h>
+#include <ros/ros.h>
+#include <string>
+#include <vector>
+
+/*
+
+Due to necessity, I had to change the PID.hh file's definition from private members to public to allow public access of its members. They're private in 1.9 and getter functions aren't implemented until gazebo 3.0. I thought that was silly, and so I hacked around it. The functions directly access the private members by necessity. It can only change if Gazebo patches 1.9 and 2.2 to include getters for it.
+
+I'm not sure exactly where the dependency chain includes PID.hh for the first time, so I've encapsulated all of the gazebo includes. Not pretty, but it works. If you're reading this and know of a better soln', feel free to change it.
+
+*/
+#define private public 
+#include <gazebo/common/Plugin.hh>
+#include <gazebo/common/Time.hh>
+#include <gazebo/math/Angle.hh>
+#include <gazebo/physics/physics.hh>
+#include <robotiq_s_model_articulated_gazebo_plugins/RobotiqHandPlugin.h>
+#undef private
+
+// Default topic names initialization.
+const std::string RobotiqHandPlugin::DefaultLeftTopicCommand  =
+  "/left_hand/command";
+const std::string RobotiqHandPlugin::DefaultLeftTopicState    =
+  "/left_hand/state";
+const std::string RobotiqHandPlugin::DefaultRightTopicCommand =
+  "/right_hand/command";
+const std::string RobotiqHandPlugin::DefaultRightTopicState   =
+  "/right_hand/state";
+
+////////////////////////////////////////////////////////////////////////////////
+RobotiqHandPlugin::RobotiqHandPlugin()
+{
+  // PID default parameters.
+  for (int i = 0; i < this->NumJoints; ++i)
+  {
+    this->posePID[i].Init(1.0, 0, 0.5, 0.0, 0.0, 60.0, -60.0);
+    this->posePID[i].SetCmd(0.0);
+  }
+
+  // Default grasping mode: Basic mode.
+  this->graspingMode = Basic;
+
+  // Default hand state: Disabled.
+  this->handState = Disabled;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+RobotiqHandPlugin::~RobotiqHandPlugin()
+{
+  gazebo::event::Events::DisconnectWorldUpdateBegin(this->updateConnection);
+  this->rosNode->shutdown();
+  this->rosQueue.clear();
+  this->rosQueue.disable();
+  this->callbackQueueThread.join();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void RobotiqHandPlugin::Load(gazebo::physics::ModelPtr _parent,
+                             sdf::ElementPtr _sdf)
+{
+  this->model = _parent;
+  this->world = this->model->GetWorld();
+  this->sdf = _sdf;
+
+  if (!this->sdf->HasElement("side") ||
```

Comment:
@TheDash This is where the implementation of "side" and assignment of static prefixes begins.  I think just requiring "prefix" would be better since that is how it is defined in the xacro